README.md

```diff
@@ -550,6 +550,55 @@ meant to be able to change with it.
     end
     ```
 
+  * <a name="let-and-let!"></a>
+    Use `let` and `let!` blocks instead of assigning values to instance
+    variables in `before` blocks.
+    <sup>[[link](#let-and-let!)]</sup>
+
+    ```ruby
+    # bad
+    describe '#type_id' do
+      before do
+        @resource = FactoryBot.create(:device)
+        @type = Type.find(@resource.type_id)
+      end
+
+      it 'sets the type_id field' do
+        expect(@resource.type_id).to equal(@type.id)
+      end
+    end
+
+    # good
+    describe '#type_id' do
+      let(:resource) { FactoryBot.create(:device) }
+      let(:type) { Type.find resource.type_id }
+
+      it 'sets the type_id field' do
+        expect(resource.type_id).to equal(type.id)
+      end
+    end
+    ```
+
+    Use `let` to initialize actions that are lazy loaded to test your specs.
+
+    ```ruby
+    context 'when updates a non-existent property value' do
+      let(:properties) { { id: 1, slected: 'on'} }
+
+      def update
+        resource.properties = properties
+      end
+
+      it 'raises an error' do
+        expect { update }.to raise_error InvalidProperty, '`slected` is not a property'
+      end
+    end
+    ```
+
+    Use `let!` to define variables even if they are not referenced in
+    examples. This can be useful to populate your database to test
+    negative cases.
```

Comment:
Perhaps we should add a warning about [mystery guest](https://robots.thoughtbot.com/mystery-guest) here, when talking about `let` and `let!`.

Reply:
Personally, I am not a fan of `let!`. It’s very, very easy to end up with unnecessary `let!`s at the top of your test file, causing all of the tests in a file to be slower than they need to be.

Reply:
Me neither, this was mostly copied over from the original. I'm all for improving this section.